webapp/src/main/webapp/js/siteAdmin/cssUploadUtils.js

```diff
@@ -46,29 +46,119 @@
     }
 
     document.addEventListener('DOMContentLoaded', function() {
-        document.getElementById('fileUpload').addEventListener('change', function(event) {
-            const file = event.target.files[0];
-            if (file && file.type === 'text/css') {
-                const reader = new FileReader();
-                reader.onload = function(e) {
-                    let styleTag = document.getElementById('dynamic-uploaded-css');
-                    if (!styleTag) {
-                        styleTag = document.createElement('style');
-                        styleTag.id = 'dynamic-uploaded-css';
-                        document.head.appendChild(styleTag);
-                    }
-                    styleTag.textContent = e.target.result;
-                };
-                reader.readAsText(file);
-            }
+        initializeFileUpload();
+        initializeFormSubmission();
+        setFormEncoding();
+    });
+
+    function initializeFileUpload() {
+        const fileInput = document.getElementById('fileUpload');
+        if (!fileInput) return;
+
+        fileInput.addEventListener('change', handleFileSelection);
+    }
+
+    function handleFileSelection(event) {
+        const file = event.target.files[0];
+        const fileInput = event.target;
+        
+        removeClearButton();
+        
+        if (file && file.type === 'text/css') {
+            createClearButton(fileInput);
+            previewCssFile(file);
+        }
+    }
+
+    function removeClearButton() {
+        const existingClearBtn = document.getElementById('clearFileBtn');
+        if (existingClearBtn) {
+            existingClearBtn.remove();
+        }
+    }
+
+    function createClearButton(fileInput) {
+        const clearBtn = document.createElement('button');
+        clearBtn.id = 'clearFileBtn';
+        clearBtn.type = 'button';
+        clearBtn.innerHTML = '✕';
+        clearBtn.title = 'Remove selected file';
+        
+        // Style the clear button
+        Object.assign(clearBtn.style, {
+            marginLeft: '10px',
```

Comment:
Wouldn't it be better to move css styles into css file and assign them by class name?